File: backend/src/main/java/com/example/stamp_app/entity/Sdi12Data.java
package com.example.stamp_app.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Sdi12Data {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column
    private String sdiAddress;

    @Column(name = "volumetricWaterContent")
    private String vwc;

    @Column(name = "soilTemperature")
    private String soilTemp;

    @Column(name = "bulkRelativePermittivity")
    private String brp;

    @Column(name = "soilBulkElectricConductivity")
    private String sbec;

    @Column(name = "soilPoreWaterElectricConductivity")
    private String spwec;

    @Column(name = "gravitationalAccelerationXAxis")
    private String gax;

    @Column(name = "gravitationalAccelerationYAxis")
    private String gay;

    @Column(name = "gravitationalAccelerationZAxis")
    private String gaz;

    @JsonBackReference
    @ManyToOne
    private MeasuredDataMaster measuredDataMaster;

    @ManyToOne
    private Sensor sensor;

}

File: backend/src/main/java/com/example/stamp_app/entity/Sdi12Data.kt
package com.example.stamp_app.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import jakarta.persistence.*
import jakarta.validation.constraints.NotNull

@Entity
data class Sdi12Data(
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	val id: Long? = null,

	@Column
	val sdiAddress: @NotNull String,

	@Column(name = "volumetricWaterContent")
	val vwc: String? = null,

	@Column(name = "soilTemperature")
	val soilTemp: String? = null,

	@Column(name = "bulkRelativePermittivity")
	val brp: String? = null,

	@Column(name = "soilBulkElectricConductivity")
	val sbec: String? = null,

	@Column(name = "soilPoreWaterElectricConductivity")
	val spwec: String? = null,

	@Column(name = "gravitationalAccelerationXAxis")
	val gax: String? = null,

	@Column(name = "gravitationalAccelerationYAxis")
	val gay: String? = null,

	@Column(name = "gravitationalAccelerationZAxis")
	val gaz: String? = null,

	@JsonBackReference
	@ManyToOne
	val measuredDataMaster: MeasuredDataMaster? = null,

	@ManyToOne
	val sensor: Sensor? = null
)
